kyo-core/shared/src/main/scala/kyo/StreamCoreExtensions.scala

```diff
@@ -69,28 +75,60 @@ object StreamCoreExtensions:
           * @param chunkSize
           *   Size of the chunks that the iterator will produce and the stream will read from
           */
-        def fromIterator[V, S](v: => Iterator[V], chunkSize: Int = Stream.DefaultChunkSize)(using
+        def fromIterator[V](v: => Iterator[V], chunkSize: Int = Stream.DefaultChunkSize)(using
             Tag[Emit[Chunk[V]]],
             Frame
         ): Stream[V, IO] =
-            Stream:
-                IO:
-                    val it      = v
-                    val size    = chunkSize max 1
-                    val builder = ChunkBuilder.init[V]
 
-                    Stream.repeatPresent(
-                        IO:
+            def toPanic[S, E <: Throwable, A](x: => A < (S & Abort[E])) =
+                Abort.recoverOrThrow(Abort.panic)(x)
+
+            fromIteratorCatching[Throwable](v, chunkSize).handle(toPanic)
```

Comment:
Isn't this the correct approach?
```suggestion
            fromIteratorCatching[Nothing](v, chunkSize)
```

Reply:
Indeed, but with a private method.